vlm/src/main/scala/geotrellis/contrib/vlm/avro/GeotrellisRasterSource.scala

```diff
@@ -35,7 +35,12 @@ import geotrellis.raster.{MultibandTile, Tile}
   * @param layerId source layer from above catalog
   * @param bandCount number of bands for each tile in above layer
   */
-case class GeotrellisRasterSource(uri: String, layerId: LayerId, bandCount: Int = 1) extends RasterSource {
+case class GeotrellisRasterSource(
+  uri: String,
+  layerId: LayerId,
+  bandCount: Int = 1,
+  private[vlm] val targetCellType: Option[TargetCellType] = None
```

Comment:
What are the effects of having this private?
Does scala create a whole new constructor for it or does the whole constructor becomes usable only by `vlm` package? Why do you think it should be private in general ?

Reply:
From both my own understanding and [this StackOverflow answer](https://stackoverflow.com/a/14694858/8469382), it looks like making `targetCellType` `val private` is the same as making it `val` just that the getter for that value becomes `private`. I'd assume then that this getter's scope is set to the `vlm` package.

I think it should be private because it seems like a piece of plumbing that the user doesn't really need to access.